### ISSUE TITLE
label parameter for Keypoint

### ISSUE THREAD

**Issue by ROCKMOHAN** [aleju/imgaug#254]
Need Optional parameter for Keypoint class as we have one for [BoundingBox](https://imgaug.readthedocs.io/en/latest/source/api_imgaug.html#imgaug.imgaug.BoundingBox) 

In case, we want to track the Keypoint transformations and pickout the labels which we want to use, It would be useful if we add label parameter.

As of now, we can do something like the below code in BB's :
`ia.BoundingBox(
x1 : 45,
y1 : 47,
x2: 52,
y2:58,
label : 'eye'
)`

How to achieve this with Keypoint Class ?

**Comment by aleju** [comment]
For keypoints that feature is not yet available. The best you can do currently is to keep a separate list of the labels and re-associate them after augmentation as the number keypoints and their indices within lists is not going to change, only their coordinates.

**Comment by ROCKMOHAN** [comment]
Sure @aleju I am now doing it as you told. Shall I implement keypoint with label, once i have time?

**Comment by aleju** [comment]
You can add a label to the keypoint class, but it will not be very useful by itself, because many augmenters re-instantiate keypoints. That means that all augmenters that create keypoints to change them also have to be changed so that they copy the label to the new class (if there is a label). 
Alternatively, they could be adjusted to all call `Keypoint.deepcopy()` (instead of __init__) with possibly changed attributes as arguments and then `deepcopy()` deals with copying the label.

**Comment by ValerianGonnot** [comment]
Yeah it could be very useful to be able to add custom attributes like label or visibility in a keypoint instance.
